packages/web/src/modules/zoom/components/zoom-provider.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { useEffect, useState, useRef, RefObject } from 'react';
+import { hooks } from '@application';
+import { Components } from '@types';
+import { User, ZoomRole } from '@chess-tent/models';
+
+import { zoomAuthorize, generateSignature } from '../requests';
+import { ZoomConnectionStatus, ZoomContext, ZoomContextType } from '../context';
+
+const { useApi, useQuery } = hooks;
+
+interface InitialContextData {
+  meetingNumber: string | undefined;
+  user: User;
+  code: string | undefined;
+  redirectUri: string;
+  zoomSDKElementRef: RefObject<HTMLElement>;
+}
+
+const createInitialContext = ({
+  meetingNumber,
+  user,
+  code,
+  redirectUri,
+  zoomSDKElementRef,
+}: InitialContextData) => ({
+  userSignature: null,
+  hostUserZakToken: undefined,
+  meetingNumber: meetingNumber?.replaceAll(' ', ''),
+  username: user.nickname,
+  password: null,
+  role: user?.coach ? ZoomRole.Host : ZoomRole.Guest,
+  authCode: code,
+  redirectUri,
+  updateContext: () => {},
+  resetContext: () => {},
+  connectionStatus: ZoomConnectionStatus.NOT_CONNECTED,
+  zoomSDKElementRef,
+});
+
+const ZoomProvider: Components['ZoomProvider'] = ({
+  redirectUri,
+  user,
+  meetingNumber,
+  children,
+}) => {
+  const {
+    fetch: authFetch,
+    response: authResponse,
+    loading: authLoading,
+  } = useApi(zoomAuthorize);
+  const {
+    fetch: signatureFetch,
+    response: signatureResponse,
+    loading: signatureLoading,
+  } = useApi(generateSignature);
+
+  const { code } = useQuery<{ code?: string }>();
+  const zoomSDKElementRef = useRef<HTMLElement>(null);
+
+  const [zoomContextState, setZoomContextState] = useState<ZoomContextType>({
+    ...createInitialContext({
+      meetingNumber,
+      user,
+      code,
+      redirectUri,
+      zoomSDKElementRef,
+    }),
+    resetContext: () =>
+      setZoomContextState(({ resetContext, updateContext }) => ({
+        ...createInitialContext({
+          meetingNumber,
+          user,
+          code,
+          redirectUri,
+          zoomSDKElementRef,
+        }),
+        resetContext,
+        updateContext,
+      })),
+    updateContext: (data: Partial<ZoomContextType>) =>
+      setZoomContextState(prevState => ({
+        ...prevState,
+        ...data,
+      })),
+  });
+
+  useEffect(() => {
+    if (!zoomContextState.authCode && zoomContextState.role === ZoomRole.Host) {
+      return;
+    } else if (zoomContextState.authCode && !authLoading && !authResponse) {
```

Comment:
You don't need `if` and `else if`. This is the perfect case for the early return pattern.

If it's anything else but what you need return early, otherwise execute your logic.